pandas/core/indexing.py

```diff
@@ -165,7 +166,28 @@ def _get_loc(self, key: int, axis: int):
     def _slice(self, obj, axis: int, kind=None):
         return self.obj._slice(obj, axis=axis, kind=kind)
 
+    def _ensure_listlike_indexer(self, key, axis):
+        # if `key` is an index to multiple columns by name, add each column
+        # if it does not already exist
+        if not isinstance(self.obj._get_axis(axis), ABCMultiIndex) and all(
+            is_hashable(k) for k in key
+        ):
+            for k in key:
+                try:
+                    self.obj[k]
+                except KeyError:
+                    self.obj[k] = np.nan
+
     def _get_setitem_indexer(self, key):
+        if (
+            self.name == "loc"  # column is indexed by name
+            and isinstance(key, tuple)
+            and len(key) >= 2  # key is at least 2-dimensional
```

Comment:
you have a lot of condictions here, when are you trying to ensure this is being called?

Reply:
`self.name == "loc"` ensures that this indexer is is indexed by name (yes for `loc`, no for `ix`).
`isinstance(key, tuple) and len(key) >= 2` filter out indexers with only 1 dimension.
`is_list_like_indexer(key[1])` ensures that the indexer is indexed by multiple columns.
`not com.is_bool_indexer(key[1])` ensures that the indexer is not a boolean indexer.

Reply:
then do this in the _LocIndexer (and call super); we don't want to dispatch on self.name like this (and .ix is going away shortly anyways).

Reply:
bigger questions, is why is this needed if you are *already* doing this for frames above.

Exactly which test hits this?

Reply:
> then do this in the _LocIndexer (and call super); we don't want to dispatch on self.name like this (and .ix is going away shortly anyways).

Done.

Reply:
> bigger questions, is why is this needed if you are _already_ doing this for frames above.
> 
> Exactly which test hits this?

See below.